### DISCUSSION TITLE
about  the using of the serviceAccount field

### DISCUSSION THREAD

**Discussion by maodia** [eclipse/jkube#2167] [Q&A] [answered]
i am using the kubernetes-maven-plugin v1.12.0.  It said "If you don’t want to generate ServiceAccount but just use an existing ServiceAccount in your Deployment. You can configure it via serviceAccount field in resource configuration"  in the doc;

here is my resouces xml config:

                     <resources>
                        <namespace>mgsc-sit</namespace>
                        <serviceAccount>mgsc-service-account</serviceAccount>
                        
                        <controller>
                            <controllerName>demo4k8s-service-consumer</controllerName>
                            <replicas>1</replicas>
                            <imagePullPolicy>Always</imagePullPolicy>
                            <liveness>
                                <getUrl>http://:8080/actuator/health/liveness</getUrl>
                                <initialDelaySeconds>20</initialDelaySeconds>
                                <timeoutSeconds>1</timeoutSeconds>
                            </liveness>
                            <readiness>
                                <getUrl>http://:8080/actuator/health/readiness</getUrl>
                                <initialDelaySeconds>15</initialDelaySeconds>
                                <timeoutSeconds>1</timeoutSeconds>
                                <failureThreshold>3</failureThreshold>
                            </readiness>
                        </controller>
                        <services>
                            <service>
                                <name>consumer-service</name>
                                <type>ClusterIP</type>
                                <ports>
                                    <port>
                                        <name>http</name>
                                        <protocol>tcp</protocol>
                                        <port>9090</port>
                                        <targetPort>8080</targetPort>
                                    </port>
                                </ports>
                            </service>
                        </services>
                    </resources>

and here is my deployment resource  yml,  i can not find the serviceAccountName config in it, so plz tell me why and what should i do ? tks

  apiVersion: apps/v1
  kind: Deployment
  metadata:
    annotations:
      jkube.eclipse.org/git-commit: 6b2e61b08fc111e7c37f36fa5b0e43d03a82e35b
      jkube.eclipse.org/git-branch: master
    labels:
      app: demo4k8s-service-consumer
      provider: jkube
      version: 1.0.0-SNAPSHOT
    name: demo4k8s-service-consumer
    namespace: mgsc-sit
  spec:
    replicas: 1
    revisionHistoryLimit: 2
    selector:
      matchLabels:
        app: demo4k8s-service-consumer
        provider: jkube
    template:
      metadata:
        annotations:
          jkube.eclipse.org/git-commit: 6b2e61b08fc111e7c37f36fa5b0e43d03a82e35b
          jkube.eclipse.org/git-branch: master
        labels:
          app: demo4k8s-service-consumer
          provider: jkube
          version: 1.0.0-SNAPSHOT
        name: demo4k8s-service-consumer
        namespace: mgsc-sit
      spec:
        containers:
        - env:
          - name: KUBERNETES_NAMESPACE
            valueFrom:
              fieldRef:
                fieldPath: metadata.namespace
          - name: HOSTNAME
            valueFrom:
              fieldRef:
                fieldPath: metadata.name
          image: registry.xxx.com/mgsc/demo4k8s-service-consumer:latest
          imagePullPolicy: Always
          livenessProbe:
            httpGet:
              host: ""
              path: /actuator/health/liveness
              port: 8080
              scheme: HTTP
            initialDelaySeconds: 20
            timeoutSeconds: 1
          name: mgsc-demo4k8s-service-consumer
          ports:
          - containerPort: 8080
            name: http
            protocol: TCP
          readinessProbe:
            failureThreshold: 3
            httpGet:
              host: ""
              path: /actuator/health/readiness
              port: 8080
              scheme: HTTP
            initialDelaySeconds: 15
            timeoutSeconds: 1
          securityContext:
            privileged: false

`

**Comment by rohanKanojia** [comment]
@maodia : Could you please edit your configuration again? Somehow it's not rendered correctly.

**Comment by rohanKanojia** [comment, marked as answer]
Looks related to https://github.com/eclipse/jkube/issues/1758 . Are you using `deployment.yaml` fragment in your application or relying on jkube defaults?

**Reply by maodia** [reply]
@rohanKanojia thanks for your answer，will it be fixed in the version 1.13.0?

**Reply by rohanKanojia** [reply]
I have a PR opened for this https://github.com/eclipse/jkube/pull/1889 . I think it should be included in 1.13.0.

If you have time, you can try building a SNAPSHOT version based on my branch and verify whether it's working as per your expectations or not.

**Reply by rohanKanojia** [reply]
@maodia : We have released [v1.13.0](https://repo1.maven.org/maven2/org/eclipse/jkube/kubernetes-maven-plugin/1.13.0/) which should contain fix for your ServiceAccount issue. Could you please give it a try and provide feedback?

**Comment by maodia** [comment]
thank you.  i tried the v1.13.1 and i am sure the problem has been fixed.

**Reply by rohanKanojia** [reply]
@maodia: Glad to hear that! Thanks a lot for trying it out and providing feedback.
